<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>Sanju&#39;s duckweed_ST35</title><link href='https://fonts.googleapis.com/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext' rel='stylesheet' type='text/css' /><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color: #ffffff; --text-color: #333333; --select-text-bg-color: #B5D6FC; --select-text-font-color: auto; --monospace: "Lucida Console",Consolas,"Courier",monospace; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; inset: 0px; font-size: 1rem; line-height: 1.428571; overflow-x: hidden; background: inherit; tab-size: 4; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; overflow-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 40px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
@media screen and (max-width: 500px) {
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:last-child { margin-bottom: 0.5rem; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; }
button, input, select, textarea { color: inherit; font-style: inherit; font-variant-caps: inherit; font-weight: inherit; font-stretch: inherit; font-size: inherit; line-height: inherit; font-family: inherit; font-size-adjust: inherit; font-kerning: inherit; font-variant-alternates: inherit; font-variant-ligatures: inherit; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-position: inherit; font-feature-settings: inherit; font-optical-sizing: inherit; font-variation-settings: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
p { line-height: inherit; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 2; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px !important; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right-width: 0px; border-right-style: none; border-right-color: currentcolor; background-color: inherit; }
.CodeMirror-linenumber { }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background: inherit; position: relative !important; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background: 0px 0px; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print {
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { print-color-adjust: exact; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 32px; padding-right: 32px; padding-bottom: 0px; break-after: avoid; }
  .typora-export #write::after { height: 0px; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: medium; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) {
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; vertical-align: initial; }
a.md-print-anchor { white-space: pre !important; border-width: medium !important; border-style: none !important; border-color: currentcolor !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; background: 0px 0px !important; text-decoration: initial !important; text-shadow: initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: hidden; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; zoom: 90%; font-size-adjust: none; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; overflow-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { transition: none; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; margin-top: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="mermaid"] svg, [lang="flow"] svg { max-width: 100%; height: auto; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom-width: 0px; border-bottom-style: none; border-bottom-color: currentcolor; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: medium; }
.highlight td, .highlight tr { border: 0px; }
svg[id^="mermaidChart"] { line-height: 1em; }


.CodeMirror { height: auto; }
.CodeMirror.cm-s-inner { background: inherit; }
.CodeMirror-scroll { overflow: auto hidden; z-index: 3; }
.CodeMirror-gutter-filler, .CodeMirror-scrollbar-filler { background-color: rgb(255, 255, 255); }
.CodeMirror-gutters { border-right-width: 1px; border-right-style: solid; border-right-color: rgb(221, 221, 221); background: inherit; white-space: nowrap; }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: rgb(0, 0, 0); }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-property { color: rgb(0, 0, 0); }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: rgb(0, 0, 255); }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: 700; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: red; }
.cm-invalidchar { color: red; }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background: inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { height: 100%; outline: 0px; position: relative; box-sizing: content-box; background: inherit; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-gutter-filler, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-vscrollbar { position: absolute; z-index: 6; display: none; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow: hidden; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow: hidden; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 30px; z-index: 3; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-wrapper { position: absolute; z-index: 4; background: 0px 0px !important; border: medium !important; }
.CodeMirror-gutter-background { position: absolute; top: 0px; bottom: 0px; z-index: 4; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-radius: 0px; border-width: 0px; background: 0px 0px; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; overflow-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; }
.CodeMirror-wrap pre { overflow-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right-width: 30px; border-right-style: solid; border-right-color: transparent; width: fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right-width: medium; border-right-style: none; border-right-color: currentcolor; width: auto; }
.CodeMirror-linebackground { position: absolute; inset: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right-width: medium; border-right-style: none; border-right-color: currentcolor; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.cm-searching { background: rgba(255, 255, 0, 0.4); }
@media print {
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}


:root {
    --side-bar-bg-color: #fafafa;
    --control-text-color: #777;
}

@include-when-export url(https://fonts.googleapis.com/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext);

/* open-sans-regular - latin-ext_latin */
  /* open-sans-italic - latin-ext_latin */
    /* open-sans-700 - latin-ext_latin */
    /* open-sans-700italic - latin-ext_latin */
  html {
    font-size: 16px;
    -webkit-font-smoothing: antialiased;
}

body {
    font-family: "Open Sans","Clear Sans", "Helvetica Neue", Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
    color: rgb(51, 51, 51);
    line-height: 1.6;
}

#write {
    max-width: 860px;
  	margin: 0 auto;
  	padding: 30px;
    padding-bottom: 100px;
}

@media only screen and (min-width: 1400px) {
	#write {
		max-width: 1024px;
	}
}

@media only screen and (min-width: 1800px) {
	#write {
		max-width: 1200px;
	}
}

#write > ul:first-child,
#write > ol:first-child{
    margin-top: 30px;
}

a {
    color: #4183C4;
}
h1,
h2,
h3,
h4,
h5,
h6 {
    position: relative;
    margin-top: 1rem;
    margin-bottom: 1rem;
    font-weight: bold;
    line-height: 1.4;
    cursor: text;
}
h1:hover a.anchor,
h2:hover a.anchor,
h3:hover a.anchor,
h4:hover a.anchor,
h5:hover a.anchor,
h6:hover a.anchor {
    text-decoration: none;
}
h1 tt,
h1 code {
    font-size: inherit;
}
h2 tt,
h2 code {
    font-size: inherit;
}
h3 tt,
h3 code {
    font-size: inherit;
}
h4 tt,
h4 code {
    font-size: inherit;
}
h5 tt,
h5 code {
    font-size: inherit;
}
h6 tt,
h6 code {
    font-size: inherit;
}
h1 {
    font-size: 2.25em;
    line-height: 1.2;
    border-bottom: 1px solid #eee;
}
h2 {
    font-size: 1.75em;
    line-height: 1.225;
    border-bottom: 1px solid #eee;
}

/*@media print {
    .typora-export h1,
    .typora-export h2 {
        border-bottom: none;
        padding-bottom: initial;
    }

    .typora-export h1::after,
    .typora-export h2::after {
        content: "";
        display: block;
        height: 100px;
        margin-top: -96px;
        border-top: 1px solid #eee;
    }
}*/

h3 {
    font-size: 1.5em;
    line-height: 1.43;
}
h4 {
    font-size: 1.25em;
}
h5 {
    font-size: 1em;
}
h6 {
   font-size: 1em;
    color: #777;
}
p,
blockquote,
ul,
ol,
dl,
table{
    margin: 0.8em 0;
}
li>ol,
li>ul {
    margin: 0 0;
}
hr {
    height: 2px;
    padding: 0;
    margin: 16px 0;
    background-color: #e7e7e7;
    border: 0 none;
    overflow: hidden;
    box-sizing: content-box;
}

li p.first {
    display: inline-block;
}
ul,
ol {
    padding-left: 30px;
}
ul:first-child,
ol:first-child {
    margin-top: 0;
}
ul:last-child,
ol:last-child {
    margin-bottom: 0;
}
blockquote {
    border-left: 4px solid #dfe2e5;
    padding: 0 15px;
    color: #777777;
}
blockquote blockquote {
    padding-right: 0;
}
table {
    padding: 0;
    word-break: initial;
}
table tr {
    border: 1px solid #dfe2e5;
    margin: 0;
    padding: 0;
}
table tr:nth-child(2n),
thead {
    background-color: #f8f8f8;
}
table th {
    font-weight: bold;
    border: 1px solid #dfe2e5;
    border-bottom: 0;
    margin: 0;
    padding: 6px 13px;
}
table td {
    border: 1px solid #dfe2e5;
    margin: 0;
    padding: 6px 13px;
}
table th:first-child,
table td:first-child {
    margin-top: 0;
}
table th:last-child,
table td:last-child {
    margin-bottom: 0;
}

.CodeMirror-lines {
    padding-left: 4px;
}

.code-tooltip {
    box-shadow: 0 1px 1px 0 rgba(0,28,36,.3);
    border-top: 1px solid #eef2f2;
}

.md-fences,
code,
tt {
    border: 1px solid #e7eaed;
    background-color: #f8f8f8;
    border-radius: 3px;
    padding: 0;
    padding: 2px 4px 0px 4px;
    font-size: 0.9em;
}

code {
    background-color: #f3f4f4;
    padding: 0 2px 0 2px;
}

.md-fences {
    margin-bottom: 15px;
    margin-top: 15px;
    padding-top: 8px;
    padding-bottom: 6px;
}


.md-task-list-item > input {
  margin-left: -1.3em;
}

@media print {
    html {
        font-size: 13px;
    }
    pre {
        page-break-inside: avoid;
        word-wrap: break-word;
    }
}

.md-fences {
	background-color: #f8f8f8;
}
#write pre.md-meta-block {
	padding: 1rem;
    font-size: 85%;
    line-height: 1.45;
    background-color: #f7f7f7;
    border: 0;
    border-radius: 3px;
    color: #777777;
    margin-top: 0 !important;
}

.mathjax-block>.code-tooltip {
	bottom: .375rem;
}

.md-mathjax-midline {
    background: #fafafa;
}

#write>h3.md-focus:before{
	left: -1.5625rem;
	top: .375rem;
}
#write>h4.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h5.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h6.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
.md-image>.md-meta {
    /*border: 1px solid #ddd;*/
    border-radius: 3px;
    padding: 2px 0px 0px 4px;
    font-size: 0.9em;
    color: inherit;
}

.md-tag {
    color: #a7a7a7;
    opacity: 1;
}

.md-toc { 
    margin-top:20px;
    padding-bottom:20px;
}

.sidebar-tabs {
    border-bottom: none;
}

#typora-quick-open {
    border: 1px solid #ddd;
    background-color: #f8f8f8;
}

#typora-quick-open-item {
    background-color: #FAFAFA;
    border-color: #FEFEFE #e5e5e5 #e5e5e5 #eee;
    border-style: solid;
    border-width: 1px;
}

/** focus mode */
.on-focus-mode blockquote {
    border-left-color: rgba(85, 85, 85, 0.12);
}

header, .context-menu, .megamenu-content, footer{
    font-family: "Segoe UI", "Arial", sans-serif;
}

.file-node-content:hover .file-node-icon,
.file-node-content:hover .file-node-open-state{
    visibility: visible;
}

.mac-seamless-mode #typora-sidebar {
    background-color: #fafafa;
    background-color: var(--side-bar-bg-color);
}

.md-lang {
    color: #b4654d;
}

/*.html-for-mac {
    --item-hover-bg-color: #E6F0FE;
}*/

#md-notification .btn {
    border: 0;
}

.dropdown-menu .divider {
    border-color: #e5e5e5;
    opacity: 0.4;
}

.ty-preferences .window-content {
    background-color: #fafafa;
}

.ty-preferences .nav-group-item.active {
    color: white;
    background: #999;
}

.menu-item-container a.menu-style-btn {
    background-color: #f5f8fa;
    background-image: linear-gradient( 180deg , hsla(0, 0%, 100%, 0.8), hsla(0, 0%, 100%, 0)); 
}



</style>
</head>
<body class='typora-export' >
<div  id='write'  class = 'is-mac'><h3><a name="rna-seq-analysis-log-for-sanjus-project" class="md-header-anchor"></a><span>RNA-Seq analysis log for Sanju&#39;s project</span></h3><p><strong><span>Project 1- Duckweed UVB Stress Experiment</span></strong></p><p><span>In this project, Sanju investigated the effects of UVB stress on duckweed plants, following a methodology similar to that used in Jie’s paper. The treatment involved exposing duckweed to different UVB exposure durations of 1, 2.5, 5, and 10 minutes, followed by different recovery intervals: 0 hours (no recovery), 12 hours, and 24 hours (recovery) , resulting in a total of 39 samples (13 groups with 3 replicates each).</span></p><p><span>To assess DNA damage and repair, more we utilized CPD (cyclobutane pyrimidine dimer) and 6-4PP (6-4 photoproduct) enzymes. The presence of bands indicated DNA damage, with red bands representing the damage. These enzymes specifically bind to cyclobutane thymidine dimers or photoproducts, allowing us to visualize the extent of DNA damage.</span></p><p><span>Sanju used the same plants for both DNA and RNA extraction, ensuring consistency across our samples. Each sample consisted of a group of plantlets, with each plantlet comprising 2-3 fronds (leaves of duckweed) and roots. This comprehensive approach allowed us to monitor the effects of UVB radiation on both genetic material and overall plant health.</span></p><p><strong><span>Project 2- Arabidopsis Sulfur Transporter Knockout Study</span></strong></p><p><span>In this project, Sanju focused on Arabidopsis plants, specifically investigating the Col-0 wild type and the ST-35 sulfur transporter knockout (KO) line. This study builds on previous research conducted with camelina. The plants were treated with selenium, a heavy metal, to evaluate their stress responses.</span></p><p><span>The results showed that the transgenic lines outperformed the wild type (WT) plants under selenium treatment. Sanju identified the best-performing transgenic line, ST-35, for further analysis. This line was selected for RNA sequencing (RNA-seq) alongside the WT to gain insights into the molecular mechanisms underlying their enhanced performance.</span></p><p><span>Additionally, Sanju employed NBT (nitroblue tetrazolium) staining to detect the accumulation of superoxide, a reactive oxygen species. This staining technique allowed for the visualization of oxidative stress levels in the plants, providing further understanding of the physiological responses to heavy metal exposure.</span></p><p><span>*For more information related to this project, go to </span><a href='/Users/yu-yaliang/Documents/Lab/Sanju/duckweed&amp;arabidopsis_12June24'><span>project folder</span></a><span> or refer to Sanju&#39;s email on June 12-13, 2024.</span></p><p>&nbsp;</p><p><strong><span>Sequencing report</span></strong></p><p><span>Sequencing is complete for samples submitted to the MSU RTSF Genomics Core, project ID </span><strong><span>MUT14810 (Duckweed &amp; Arabidopsis-RNA_Illumina Seq)</span></strong><span>. This project submitted </span><strong><span>69 samples</span></strong><span> of plant total RNA submitted for NGS library preparation and sequencing. Libraries were prepared using the </span><strong><span>Roche KAPA mRNA HyperPrep Kit</span></strong><span> with KAPA UDI primers following manufacturer’s recommendations. Completed libraries were QC’d and quantified using a combination of Qubit dsDNA HS and Agilent 4200 TapeStation HS DNA1000 assays. The libraries were pooled in equimolar proportions and the pool quantified using the Invitrogen Collibri Quantification qPCR kit.</span></p><p><span>The pool was loaded onto one </span><strong><span>(1) lane of an NovaSeq S4 flow cell</span></strong><span> using the Xp loading kit and workflow. The flow cell was mounted on a </span><strong><span>NovaSeq 6000</span></strong><span> instrument and sequencing was performed in a </span><strong><span>2x150bp paired end</span></strong><span> format using a NovaSeq v1.5, 300 cycle reagent kit. Base calling was done by Illumina Real Time Analysis (RTA) v3.4.4 and output of RTA was demultiplexed and converted to FastQ format with Illumina Bcl2fastq v2.20.0. A summary of the run output is attached below. Basic QC information about your sequence data is provided by the accompanying FastQC reports. Please see the FastQC Tutorial and FAQ (</span><a href='https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Frtsf.natsci.msu.edu%2Fgenomics%2Ftechnical-documents%2Ffastqc-tutorial-and-faq.aspx&amp;data=05|02|liangyu%40uvu.edu|925ce1b2564d4cb275eb08dc8b284c49|1ea2b65f2f5e440eb025dfdfafd8e097|0|0|638538252410297745|Unknown|TWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D|0|||&amp;sdata=Mz23X1EjhwODzwMOtzW7er%2BMvW4FS9ddtzo5afUoqyQ%3D&amp;reserved=0'><span>https://rtsf.natsci.msu.edu/genomics/technical-documents/fastqc-tutorial-and-faq.aspx</span></a><span>) for information regarding interpretation of these reports.</span></p><p><span>*The data will be processed using Trimmomatic-STAR alignment-SAMTools-HTseq-count pipeline.</span></p><hr /><p>&nbsp;</p><p><strong><span>June 18, 2024</span></strong></p><ol start='' ><li><p><span>Setup working directories</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae Sanju]<span class="cm-def">$ mkdir</span> duckweed_UVB</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae Sanju]<span class="cm-def">$ mkdir</span> arabidopsis_duckweed_Sulfer_KO</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ mkdir</span> raw_reads</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ mkdir</span> trimmed</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 115px;"></div><div class="CodeMirror-gutters" style="display: none; height: 115px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Download sequencing results</span></p><p><span>FastQC wasn&#39;t provided by Sanju.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ wget</span> <span class="cm-attribute">-r</span> <span class="cm-attribute">-np</span> <span class="cm-attribute">-nH</span> <span class="cm-attribute">--ask-password</span> ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-number">2024</span><span class="cm-attribute">-06-18</span> <span class="cm-number">13</span>:49:07 (112 MB/s) <span class="cm-attribute">-</span> ‘20240223_mRNASeq_PE150/ST35-4_4-2_S65_L003_R2_001.fastq.gz’ saved [3061635507]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--2024-06-18</span> <span class="cm-number">13</span>:49:07--  ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150/ST35-4_4-3_S66_L003_R1_001.fastq.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; <span class="cm-operator">=</span>&gt; ‘20240223_mRNASeq_PE150/ST35-4_4-3_S66_L003_R1_001.fastq.gz’</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; CWD not required.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; PASV ... <span class="cm-keyword">done</span>. &nbsp; &nbsp;<span class="cm-operator">==</span>&gt; RETR ST35-4_4-3_S66_L003_R1_001.fastq.gz ... <span class="cm-keyword">done</span>.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Length: <span class="cm-number">3423342003</span> (3.2G)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Segmentation fault (core dumped)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># something wrong. didn't see md5sum file. not sure which step it terminated. </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># go to sync to archive for now.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># same error happened. Talked to Kevin on 20June24, and it turns out to be not enough storage.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## June 21, 2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># remove all downloaded files from both archive</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(new_env) [yoyopikachu@orchidaceae Sanju]<span class="cm-def">$ pwd</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">/archive/yoyopikachu/orchidaceae_archive/raw_data/Sanju</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(new_env) [yoyopikachu@orchidaceae Sanju]<span class="cm-def">$ rm</span> <span class="cm-attribute">-r</span> 20240223_mRNASeq_PE150</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ pwd</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">/data/run/yoyopikachu/Sanju/duckweed_UVB/raw_reads</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ rm</span> <span class="cm-attribute">-r</span> 20240223_mRNASeq_PE150/</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae Sanju]<span class="cm-def">$ rm</span> <span class="cm-attribute">-r</span> arabidopsis_duckweed_Sulfer_KO</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 667px;"></div><div class="CodeMirror-gutters" style="display: none; height: 667px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Download again (11Jul24)</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ wget</span> <span class="cm-attribute">-r</span> <span class="cm-attribute">-np</span> <span class="cm-attribute">-nH</span> <span class="cm-attribute">--ask-password</span> ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--2024-07-11</span> <span class="cm-number">16</span>:29:45--  ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150/ST35-4_5-1_S67_L003_R1_001.fastq.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; <span class="cm-operator">=</span>&gt; ‘20240223_mRNASeq_PE150/ST35-4_5-1_S67_L003_R1_001.fastq.gz’</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; CWD not required.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; PASV ... <span class="cm-keyword">done</span>. &nbsp; &nbsp;<span class="cm-operator">==</span>&gt; RETR ST35-4_5-1_S67_L003_R1_001.fastq.gz ... <span class="cm-keyword">done</span>.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Length: <span class="cm-number">3416308244</span> (3.2G)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Segmentation fault (core dumped)</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 230px;"></div><div class="CodeMirror-gutters" style="display: none; height: 230px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Download on </span><code>Poaceae</code><span> for now and will clean up </span><code>Orchid</code><span> later</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@poaceae yoyopikachu]<span class="cm-def">$ </span>&nbsp;<span class="cm-builtin">mkdir</span> duckweed_UVB</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@poaceae duckweed_UVB]<span class="cm-def">$ mkdir</span> raw_reads</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@poaceae raw_reads]<span class="cm-def">$ module</span> load <span class="cm-builtin">wget</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@poaceae raw_reads]<span class="cm-def">$ wget</span> <span class="cm-attribute">-r</span> <span class="cm-attribute">-np</span> <span class="cm-attribute">-nH</span> <span class="cm-attribute">--ask-password</span> ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-number">2024</span><span class="cm-attribute">-07-12</span> <span class="cm-number">12</span>:41:09 (112 MB/s) <span class="cm-attribute">-</span> ‘20240223_mRNASeq_PE150/ST35-4_4-3_S66_L003_R1_001.fastq.gz’ saved [3423342003]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--2024-07-12</span> <span class="cm-number">12</span>:41:09--  ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150/ST35-4_4-3_S66_L003_R2_001.fastq.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; <span class="cm-operator">=</span>&gt; ‘20240223_mRNASeq_PE150/ST35-4_4-3_S66_L003_R2_001.fastq.gz’</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; CWD not required.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; PASV ... <span class="cm-keyword">done</span>. &nbsp; &nbsp;<span class="cm-operator">==</span>&gt; RETR ST35-4_4-3_S66_L003_R2_001.fastq.gz ... <span class="cm-keyword">done</span>.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Length: <span class="cm-number">3585527718</span> (3.3G)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Segmentation fault (core dumped)</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 414px;"></div><div class="CodeMirror-gutters" style="display: none; height: 414px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>(12Jul24) Download to my NAS </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">liangyu@YuYaLiangNAS:/volume1/Research/raw_reads/sanju<span class="cm-def">$ pwd</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">/volume1/Research/raw_reads/sanju</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">liangyu@YuYaLiangNAS:/volume1/Research/raw_reads/sanju<span class="cm-def">$ wget</span> <span class="cm-attribute">-r</span> <span class="cm-attribute">-np</span> <span class="cm-attribute">-nH</span> <span class="cm-attribute">--ask-password</span> ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 15JUl24 </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># looks like files are downloaded completely.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># md5sum check</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">md5sum *.gz &gt; md5_NAS_15Jul24.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># md5 results are the same with the order of "control1" different.</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 299px;"></div><div class="CodeMirror-gutters" style="display: none; height: 299px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>(22Jul24) After meet with Kevin last Thursday, Orchid should have enough space. Download raw read again.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># close previous screen and reopen</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ module</span> load <span class="cm-builtin">wget</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ wget</span> <span class="cm-attribute">-r</span> <span class="cm-attribute">-np</span> <span class="cm-attribute">-nH</span> <span class="cm-attribute">--ask-password</span> ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#username: &nbsp; &nbsp; &nbsp; sanjaya</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#password: &nbsp; &nbsp; &nbsp; Y2c4ahLH</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#### error</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">20240223_mRNASeq_PE150/ST35-4 <span class="cm-number">100</span>%[<span class="cm-operator">=================================================</span>&gt;] &nbsp; <span class="cm-number">3</span>.34G &nbsp; 112MB/s &nbsp; &nbsp;<span class="cm-keyword">in</span> 31s</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-number">2024</span><span class="cm-attribute">-07-22</span> <span class="cm-number">15</span>:06:00 (110 MB/s) <span class="cm-attribute">-</span> ‘20240223_mRNASeq_PE150/ST35-4_4-3_S66_L003_R2_001.fastq.gz’ saved [3585527718]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--2024-07-22</span> <span class="cm-number">15</span>:06:00--  ftps://sanjaya@titan.bch.msu.edu/20240223_mRNASeq_PE150/ST35-4_5-1_S67_L003_R1_001.fastq.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; <span class="cm-operator">=</span>&gt; ‘20240223_mRNASeq_PE150/ST35-4_5-1_S67_L003_R1_001.fastq.gz’</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; CWD not required.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-operator">==</span>&gt; PASV ... <span class="cm-keyword">done</span>. &nbsp; &nbsp;<span class="cm-operator">==</span>&gt; RETR ST35-4_5-1_S67_L003_R1_001.fastq.gz ... <span class="cm-keyword">done</span>.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Length: <span class="cm-number">3416308244</span> (3.2G)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Segmentation fault (core dumped)</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 529px;"></div><div class="CodeMirror-gutters" style="display: none; height: 529px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>(30Jul24) Kevin downloaded the files in batches on Orchid.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># remove the old ones on my directory</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ rm</span> <span class="cm-attribute">-r</span> 20240223_mRNASeq_PE150/</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># save a copy to my directory and archive</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># '/archive/yoyopikachu/orchidaceae_archive/raw_data/Sanju'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae raw_reads]<span class="cm-def">$ rsync</span> <span class="cm-attribute">-h</span> <span class="cm-attribute">-r</span> /data/run/kchilds/Sanjaya/20240223_mRNASeq_PE150 /data/run/yoyopikachu/Sanju/duckweed_UVB/raw_reads</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae Sanju]<span class="cm-def">$ rsync</span> <span class="cm-attribute">-h</span> <span class="cm-attribute">-r</span> /data/run/kchilds/Sanjaya/20240223_mRNASeq_PE150 /archive/yoyopikachu/orchidaceae_archive/raw_data/Sanju</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># md5sum check</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ md5sum</span> *.gz &gt; md5_archive_30Jul24.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ md5sum</span> *.gz &gt; md5_30Jul24.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># md5 values of both loactions are correct.</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 345px;"></div><div class="CodeMirror-gutters" style="display: none; height: 345px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Download sequencing results</span></p><p><span>FastQC wasn&#39;t provided by Sanju.</span></p></li></ol><p>&nbsp;</p><p><strong><span>Nov. 16, 2024</span></strong></p><p><span>	</span><span>Starting the analysis on Orchidaceae. Wasn&#39;t sure the adapter sequences, therefore whent back to previous note </span><code>Note_Riboseq_Kapa_mRNAseq_rice_salt.md</code><span>.  Based on the note, the adapter sequences should be </span><code>TruSeq3-SE.fa</code><span>. However, the scripts and the files in the folder </span><code>RiboSeq</code><span> can no longer be found. Below are the commands:</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># copy script from previous mRNAseq directory (20220322_mRNASeq_SE100)</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae 20220408_mRNASeq_SE100]<span class="cm-def">$ cp</span> /data/run/yoyopikachu/RiboSeq/20220322_mRNASeq_SE100/Trim_28Mar22.sh Trim_19Apr22.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae 20220408_mRNASeq_SE100]<span class="cm-def">$ ln</span> <span class="cm-attribute">-s</span> /data/run/yoyopikachu/RiboSeq/20220322_mRNASeq_SE100/TruSeq3-SE.fa</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 115px;"></div><div class="CodeMirror-gutters" style="display: none; height: 115px;"></div></div></div></pre><p>&nbsp;</p><ol start='' ><li><p><span>Trimming</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## copy scripts</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ cp</span> /data/run/yoyopikachu/Sanju/Trim/Trim_camelina.sh Trim_st35.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ cp</span> /data/run/yoyopikachu/Sanju/Trim/Trim_camelina.sh Trim_uvb.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ ln</span> <span class="cm-attribute">-s</span> /data/run/yoyopikachu/Sanju/Trim/TruSeq3-SE.fa .</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Sanju's sequencing is PE so create a new adapter seq file based on the sequences on Apiaceae</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@apiaceae yoyopikachu]<span class="cm-def">$ cd</span> Adapter_seq/</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@apiaceae Adapter_seq]<span class="cm-def">$ ls</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">TruSeq3-PE.fa  TruSeq3-SE.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ vi</span> TruSeq3-PE.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## manually copy script for trimming from Alicia's directory</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># file path: /data/run/yoyopikachu/Alicia/trim_AR.sh</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## fix naming </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ for</span> file <span class="cm-keyword">in</span> *.fastq.gz; <span class="cm-keyword">do</span> <span class="cm-builtin">mv</span> <span class="cm-string">"</span><span class="cm-def">$file</span><span class="cm-string">"</span> <span class="cm-string">"</span><span class="cm-quote">$(echo "</span><span class="cm-def">$file</span><span class="cm-quote">" | sed 's/_S[0-9][0-9]//')</span><span class="cm-string">"</span>; <span class="cm-keyword">done</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min0hr-1_S4_L003_R1_001.fastq.gz’ and ‘1min0hr-1_S4_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min0hr-1_S4_L003_R2_001.fastq.gz’ and ‘1min0hr-1_S4_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min0hr-2_S5_L003_R1_001.fastq.gz’ and ‘1min0hr-2_S5_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min0hr-2_S5_L003_R2_001.fastq.gz’ and ‘1min0hr-2_S5_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min0hr-3_S6_L003_R1_001.fastq.gz’ and ‘1min0hr-3_S6_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min0hr-3_S6_L003_R2_001.fastq.gz’ and ‘1min0hr-3_S6_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min12hr-1_S7_L003_R1_001.fastq.gz’ and ‘1min12hr-1_S7_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min12hr-1_S7_L003_R2_001.fastq.gz’ and ‘1min12hr-1_S7_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min12hr-2_S8_L003_R1_001.fastq.gz’ and ‘1min12hr-2_S8_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min12hr-2_S8_L003_R2_001.fastq.gz’ and ‘1min12hr-2_S8_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min12hr-3_S9_L003_R1_001.fastq.gz’ and ‘1min12hr-3_S9_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘1min12hr-3_S9_L003_R2_001.fastq.gz’ and ‘1min12hr-3_S9_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘control1_S1_L003_R1_001.fastq.gz’ and ‘control1_S1_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘control1_S1_L003_R2_001.fastq.gz’ and ‘control1_S1_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘control2_S2_L003_R1_001.fastq.gz’ and ‘control2_S2_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘control2_S2_L003_R2_001.fastq.gz’ and ‘control2_S2_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘control3_S3_L003_R1_001.fastq.gz’ and ‘control3_S3_L003_R1_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">mv</span>: ‘control3_S3_L003_R2_001.fastq.gz’ and ‘control3_S3_L003_R2_001.fastq.gz’ are the same file</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## change name for UVB control group so they can be trimmed using the same script</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ for</span> file <span class="cm-keyword">in</span> control*.fastq.gz; <span class="cm-keyword">do</span> <span class="cm-builtin">mv</span> <span class="cm-string">"</span><span class="cm-def">$file</span><span class="cm-string">"</span> <span class="cm-string">"</span><span class="cm-quote">$(echo "</span><span class="cm-def">$file</span><span class="cm-quote">" | sed -E 's/control([0-9])_/control-\1_/; s/_S[0-9]//')</span><span class="cm-string">"</span>; <span class="cm-keyword">done</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## copy script for ST35</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ cp</span> Trim_UVB.sh Trim_ST35.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## fixing file names for ST35 </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ for</span> file <span class="cm-keyword">in</span> ST35*.fastq.gz; <span class="cm-keyword">do</span> <span class="cm-builtin">mv</span> <span class="cm-string">"</span><span class="cm-def">$file</span><span class="cm-string">"</span> <span class="cm-string">"</span><span class="cm-quote">$(echo "</span><span class="cm-def">$file</span><span class="cm-quote">" | sed 's/-4_/_/')</span><span class="cm-string">"</span>; <span class="cm-keyword">done</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## run trimming </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ </span>. Trim_UVB.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ </span>. Trim_ST35.sh</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1288px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1288px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Generate an update GFF file with primary transcripts only. Duckweed reference sequence and annotation files were downloaded from previous duckweed project (see </span><code>Sanju&#39;s duckweed.md</code><span> for detail).</span></p><ul><li><p><span>Based on previous note, the GFF file is not primary transcript only</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Obtain primary transcript from cds.all.fa using python script</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of peptide input:20705</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of primary peptide:20659</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Scripts_py/get_primary_from_pep.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Sanju/Duckweed/protein.fasta <span class="cm-attribute">-o1</span> /Users/yoyopikachu/Documents/Lab/Sanju/Duckweed/duckweed_primary_pep.txt <span class="cm-attribute">-o2</span> /Users/yoyopikachu/Documents/Lab/Sanju/Duckweed/duckweed_primary_pep_id.txt</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 184px;"></div><div class="CodeMirror-gutters" style="display: none; height: 184px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Check whether </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed]<span class="cm-def">$ grep</span> <span class="cm-attribute">-c</span> <span class="cm-string">'&gt;'</span> duckweed_primary_pep.txt</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-number">20659</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed]<span class="cm-def">$ awk</span> <span class="cm-attribute">-F</span> <span class="cm-string">'\t'</span> <span class="cm-string">'$3 == "mRNA" {count++} END {print count}'</span> Spirodela_polyrhiza_Greater_Duckweed_strain_9509_annos1-cds0-id_typename-nu1-upa1-add_chr0.gid51364.gff</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-number">20705</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## print out diplicated genes and sequences</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed]<span class="cm-def">$ awk</span> <span class="cm-attribute">-F</span> <span class="cm-string">'\\|\\|'</span> <span class="cm-string">'/^&gt;/ {print $5}'</span> <span class="cm-number">51364</span><span class="cm-attribute">-CDS-prot</span>.fasta | <span class="cm-builtin">sort</span> | uniq <span class="cm-attribute">-d</span> | <span class="cm-keyword">while</span> read id; <span class="cm-keyword">do</span> <span class="cm-builtin">grep</span> <span class="cm-attribute">-A</span> <span class="cm-number">1</span> <span class="cm-string">"||</span><span class="cm-def">$id</span><span class="cm-string">||"</span> <span class="cm-number">51364</span><span class="cm-attribute">-CDS-prot</span>.fasta; <span class="cm-keyword">done</span> &gt; duplicated_duckweed_genes.fasta</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed]<span class="cm-def">$ grep</span> <span class="cm-attribute">-c</span> <span class="cm-string">'&gt;'</span> duplicated_duckweed_genes.fasta</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-number">2364</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## print out gene id</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed]<span class="cm-def">$ awk</span> <span class="cm-attribute">-F</span> <span class="cm-string">'\\|\\|'</span> <span class="cm-string">'/^&gt;/ {print $5}'</span> duplicated_duckweed_genes.fasta</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed]<span class="cm-def">$ awk</span> <span class="cm-attribute">-F</span> <span class="cm-string">'\\|\\|'</span> <span class="cm-string">'/^&gt;/ {print $5}'</span> duplicated_duckweed_genes.fasta | <span class="cm-builtin">tee</span> &gt;(echo <span class="cm-string">"Unique IDs: </span><span class="cm-quote">$(sort | uniq | wc -l)</span><span class="cm-string">"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Unique IDs: 418</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 483px;"></div><div class="CodeMirror-gutters" style="display: none; height: 483px;"></div></div></div></pre><blockquote><p><span>Taking these gene ID and check GFF file. Randomly chose three to check, all of them has only one &#39;gene&#39; and &#39;mRNA&#39; exist. Therefore, I think it&#39;s fine to use the current GFF file for alignment.</span></p></blockquote><p>&nbsp;</p></li></ul></li><li><p><span>Alignment</span></p><p><span>The most recent RNA-seq project with Sanju was transgenic switchgrass (see </span><code>Sanju&#39;s duckweed_ST35.md</code><span>) therefore, copy scripts from alignment from </span><code>transgenic_sg</code><span> on Orchidaceae.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ mkdir</span> STAR_alignment</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> /data/run/yoyopikachu/Sanju/transgenic_sg/Genome_index_sg_14Apr22.run Genome_index_duckweed.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> /data/run/yoyopikachu/Sanju/transgenic_sg/STAR_map_sg_14Apr22.run STAR_map_duckweed_UVB.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## genome index</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. Genome_index_duckweed.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">16</span> <span class="cm-number">22</span>:59:45 WARNING: <span class="cm-attribute">--genomeSAindexNbases</span> <span class="cm-number">13</span> is too large <span class="cm-keyword">for</span> the genome <span class="cm-def">size</span><span class="cm-operator">=</span><span class="cm-number">141557760</span>, which may cause seg-fault at the mapping step. Re-run genome generation with recommended <span class="cm-attribute">--genomeSAindexNbases</span> <span class="cm-number">12</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## update '--genomeSAindexNbases' and rerun</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. Genome_index_duckweed.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Run alignment</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## --sjdbOverhang 149</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## using PE for alignment</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## To speed up the process, run alignments in three script</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#### ST35 project</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_ST.run STAR_map_duckweed_ST_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_ST.run STAR_map_duckweed_ST_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_ST.run STAR_map_duckweed_ST_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_ST.run STAR_map_duckweed_Col_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_ST.run STAR_map_duckweed_Col_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_ST.run STAR_map_duckweed_Col_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Col01_1&amp;Col01_2</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Col01_3&amp;Col01_4</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Col01_5</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## ST35_1&amp;ST35_2</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## ST35_3&amp;ST35_4</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## ST35_5</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## regenerate genome index with --sjdbOverhang 149</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. Genome_index_duckweed.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#### UVB project</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_UVB.run STAR_map_duckweed_UVB_10.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_UVB.run STAR_map_duckweed_UVB_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_UVB.run STAR_map_duckweed_UVB_25.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_UVB.run STAR_map_duckweed_UVB_5.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ cp</span> STAR_map_duckweed_UVB.run STAR_map_duckweed_UVB_control.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## run</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_UVB_10.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_UVB_25.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_UVB_5.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(brassica) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_UVB_control.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_UVB_1.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1472px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1472px;"></div></div></div></pre><blockquote><ul><li><p><code>STAR_map_duckweed_UVB_1.run</code><span> didn&#39;t work because input files are missing.</span></p></li><li><p><span>Checked </span><code>trimmed</code><span> directory and only </span><code>1min24hr</code><span> has results. Turned out it was due to the name of tthe files. The &#39;S_&#39; were not removed from the file names.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## rename</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ for</span> file <span class="cm-keyword">in</span> 1min*.fastq.gz; <span class="cm-keyword">do</span> <span class="cm-builtin">mv</span> <span class="cm-string">"</span><span class="cm-def">$file</span><span class="cm-string">"</span> <span class="cm-string">"</span><span class="cm-quote">$(echo "</span><span class="cm-def">$file</span><span class="cm-quote">" | sed 's/_S[0-9]//')</span><span class="cm-string">"</span>; <span class="cm-keyword">done</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## copy script, edit samples, and rerun</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ cp</span> Trim_UVB.sh Trim_UVB_1min.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae 20240223_mRNASeq_PE150]<span class="cm-def">$ </span>. Trim_UVB_1min.sh</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 161px;"></div><div class="CodeMirror-gutters" style="display: none; height: 161px;"></div></div></div></pre></li></ul></blockquote></li></ol><p>&nbsp;</p><ol start='4' ><li><p><span>Sorting</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## copy script</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae STAR_alignment]<span class="cm-def">$ cp</span> /data/run/yoyopikachu/Sanju/transgenic_sg/STAR_alignment/sort_by_pos_20Apr22.sh sort_by_pos.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Test script with 'control_1_Aligned.out.sam' and 'control_2_Aligned.out.sam'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae STAR_alignment]<span class="cm-def">$ </span>. sort_by_pos.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Thu Nov 21 00:55:31 EST 2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Sorting sample: control_1</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#[bam_sort_core] merging from 35 files and 1 in-memory blocks...</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Sorting sample: control_2</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#[bam_sort_core] merging from 38 files and 1 in-memory blocks...</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Thu Nov 21 05:05:02 EST 2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Edit and run</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae STAR_alignment]<span class="cm-def">$ </span>. sort_by_pos_20Apr22.sh</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 368px;"></div><div class="CodeMirror-gutters" style="display: none; height: 368px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Terminate</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## </span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Mon Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:00 EST <span class="cm-number">2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Running STAR sample: ST35_3_1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:00 ..... started STAR run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:00 ..... loading genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:02 ..... processing annotations GTF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:07 ..... inserting junctions into the genome indices</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:14 ..... started mapping</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">^C</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">##</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Mon Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:16 EST <span class="cm-number">2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Running STAR sample: ST35_1_1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:16 ..... started STAR run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:16 ..... loading genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:18 ..... processing annotations GTF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:23 ..... inserting junctions into the genome indices</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:31 ..... started mapping</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">^C</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Mon Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:30 EST <span class="cm-number">2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Running STAR sample: Col01_5_1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:30 ..... started STAR run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:31 ..... loading genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:32 ..... processing annotations GTF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:39 ..... inserting junctions into the genome indices</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:49 ..... started mapping</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">^C</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">##</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Mon Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:40 EST <span class="cm-number">2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Running STAR sample: Col01_3_1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:40 ..... started STAR run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:41 ..... loading genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:43 ..... processing annotations GTF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:15:51 ..... inserting junctions into the genome indices</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:03 ..... started mapping</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">^C</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">##</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Mon Nov <span class="cm-number">18</span> <span class="cm-number">13</span>:03:35 EST <span class="cm-number">2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Running STAR sample: ST35_5_1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">13</span>:03:35 ..... started STAR run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">13</span>:03:35 ..... loading genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">13</span>:03:44 ..... processing annotations GTF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">13</span>:04:18 ..... inserting junctions into the genome indices</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">13</span>:04:59 ..... started mapping</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">^C</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">##</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Mon Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:09 EST <span class="cm-number">2024</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Running STAR sample: Col01_1_1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:09 ..... started STAR run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:09 ..... loading genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:12 ..... processing annotations GTF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:24 ..... inserting junctions into the genome indices</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Nov <span class="cm-number">18</span> <span class="cm-number">09</span>:16:41 ..... started mapping</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">^C</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">##</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1541px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1541px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Rerun what were terminated in Step 5 above </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Wed Dec  4 01:21:51 EST 2024</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_Col_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_1.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_2.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(base) [yoyopikachu@orchidaceae duckweed_UVB]<span class="cm-def">$ </span>. STAR_map_duckweed_ST_3.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 184px;"></div><div class="CodeMirror-gutters" style="display: none; height: 184px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>f</span></p></li></ol><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p><strong><span>November 16, 2020</span></strong></p><ol start='2' ><li><p><span>Adapter trimming</span></p><p><span>In order to make the following coding easier, parts of file name were removed for all read files. For example, &#39;C_G-3_S8_L006_R1_001.fastq.gz&#39;  became &#39;camelina_Gypsum_3_R1.fastq.gz&#39;. For switchgrass, &#39;SG-Coal-Up-40-3_S8_L003_R1_001.fastq.gz&#39; will be change to &#39;SG-Coal-Up-40-3_R1.fastq.gz&#39;. LRemoving was conducted on FileZilla.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Adapter sequences TruSeq2 (GAII) and TruSeq3 (HiSeq and MiSeq)</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># scripts were copied from carex juncus mRNA directory</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae Trim]<span class="cm-def">$ </span>. Trim_camelina.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae Trim]<span class="cm-def">$ </span>. Trim_switchgrass.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> </span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 138px;"></div><div class="CodeMirror-gutters" style="display: none; height: 138px;"></div></div></div></pre><blockquote><ul><li><span>Camelina (see &#39;Trim_camelina_log&#39; for more detail)</span></li></ul><figure><table><thead><tr><th>&nbsp;</th><th><span>Raw</span></th><th><span>After trimmed</span></th></tr></thead><tbody><tr><td><span>Agri1</span></td><td><span>36342888</span></td><td><span>35905914 (98.80%)</span></td></tr><tr><td><span>Agri2</span></td><td><span>42402147</span></td><td><span>41813044 (98.61%)</span></td></tr><tr><td><span>Agri3</span></td><td><span>41623041</span></td><td><span>41131081 (98.82%)</span></td></tr><tr><td><span>Coalmine1</span></td><td><span>38040741</span></td><td><span>37655516 (98.99%)</span></td></tr><tr><td><span>Coalmine2</span></td><td><span>46327000</span></td><td><span>46100252 (99.51%)</span></td></tr><tr><td><span>Coalmine3</span></td><td><span>47903995</span></td><td><span>47741235 (99.66%)</span></td></tr><tr><td><span>Gypsum1</span></td><td><span>38589534</span></td><td><span>37725981 (97.76%)</span></td></tr><tr><td><span>Gypsum2</span></td><td><span>34969234</span></td><td><span>34701030 (99.23%)</span></td></tr><tr><td><span>Gypsum3</span></td><td><span>31277504</span></td><td><span>30994133 (99.09%)</span></td></tr></tbody></table></figure></blockquote></li></ol><blockquote><ul><li><span>Switchgrass (see &#39;Trim_switchgrass_log&#39; for more detail)</span></li></ul><figure><table><thead><tr><th>&nbsp;</th><th><span>Raw</span></th><th><span>After trimmed</span></th></tr></thead><tbody><tr><td><span>SG-Ag-Up-40-1</span></td><td><span>26066021</span></td><td><span>25969385 (99.63%)</span></td></tr><tr><td><span>SG-Ag-Up-40-2</span></td><td><span>29285087</span></td><td><span>29181121 (99.64%)</span></td></tr><tr><td><span>SG-Ag-Up-40-3</span></td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td><span>SG-Ag-Low-42-1</span></td><td><span>31716770</span></td><td><span>31571772 (99.54%)</span></td></tr><tr><td><span>SG-Ag-Low-42-2</span></td><td><span>32478776</span></td><td><span>32074539 (98.76%)</span></td></tr><tr><td><span>SG-Ag-Low-42-3</span></td><td><span>32932442</span></td><td><span>32650266 (99.14%)</span></td></tr><tr><td><span>SG-Coal-Up-40-1</span></td><td><span>32949164</span></td><td><span>32780281 (99.49%)</span></td></tr><tr><td><span>SG-Coal-Up-40-2</span></td><td><span>24606565</span></td><td><span>24428342 (99.28%)</span></td></tr><tr><td><span>SG-Coal-Up-40-3</span></td><td><span>29678700</span></td><td><span>29568947 (99.63%)</span></td></tr><tr><td><span>SG-Coal-Low-42-1</span></td><td><span>32144247</span></td><td><span>31960681 (99.43%)</span></td></tr><tr><td><span>SG-Coal-Low-42-2</span></td><td><span>28987605</span></td><td><span>28698313 (99.00%)</span></td></tr><tr><td><span>SG-Coal-Low-42-3</span></td><td><span>28486714</span></td><td><span>28187508 (98.95%)</span></td></tr></tbody></table></figure></blockquote><p><span>   </span></p><ol start='3' ><li><p><span>Reference sequence and annotation</span></p><ul><li><p><span>Downloaded Camelina reference genome and annotation </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae ref_seq_annotation]<span class="cm-def">$ wget</span> ftp://ftp.ensemblgenomes.org/pub/plants/release-48/fasta/camelina_sativa/dna/Camelina_sativa.Cs.dna.toplevel.fa.gz</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae ref_seq_annotation]<span class="cm-def">$ gzip</span> <span class="cm-attribute">-d</span> Camelina_sativa.Cs.dna.toplevel.fa.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae ref_seq_annotation]<span class="cm-def">$ wget</span> ftp://ftp.ensemblgenomes.org/pub/plants/release-48/gff3/camelina_sativa/Camelina_sativa.Cs.48.chr.gff3.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae ref_seq_annotation]<span class="cm-def">$ gzip</span> <span class="cm-attribute">-d</span> Camelina_sativa.Cs.48.chr.gff3.gz</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 138px;"></div><div class="CodeMirror-gutters" style="display: none; height: 138px;"></div></div></div></pre><blockquote><p><span>There are three assemblies in EnsemblPlant, the difference is in the masking of the repeats:</span></p><ol start='' ><li><span>_sm = soft masking (lower case)</span></li><li><span>_rm = hard masking (Ns)</span></li><li><span>no masking</span>
<span>Hard masking should not be used for mapping in most scenarios.</span>
<span>No masking and sm should behave exactly the same with STAR, so you can choose either.</span></li></ol><p><a href='https://github.com/alexdobin/STAR/issues/479' target='_blank' class='url'>https://github.com/alexdobin/STAR/issues/479</a></p></blockquote></li><li><p><span>Switchgrass</span></p><p><span>Dwonloaded assembly (Pvirgatum: Pvirgatum_516_v5.0.fa.gz) and annotation (Pvirgatum: Pvirgatum_516_v5.1.gene.gff3.gz) from </span><a href='https://genome.jgi.doe.gov/portal/pages/dynamicOrganismDownload.jsf?organism=Pvirgatum' target='_blank' class='url'>https://genome.jgi.doe.gov/portal/pages/dynamicOrganismDownload.jsf?organism=Pvirgatum</a><span> and uploaded to server via FileZilla.</span></p><p>&nbsp;</p></li></ul></li><li><p><span>STAR alignment</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># camelina</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. Genome_index_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. STAR_map_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass]<span class="cm-def">$ </span>. Genome_index.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass]<span class="cm-def">$ </span>. STAR_map.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 161px;"></div><div class="CodeMirror-gutters" style="display: none; height: 161px;"></div></div></div></pre><blockquote><p><span>[yoyopikachu@orchidaceae switchgrass]$ . Genome_index.run</span>
<span>Mon Nov 16 16:54:05 EST 2020</span>
<span>Nov 16 16:54:05 ..... started STAR run</span>
<span>Nov 16 16:54:05 ... starting to generate Genome files</span>
<span>Nov 16 16:54:30 ..... processing annotations GTF</span></p><p><span>Fatal INPUT FILE error, no exon lines in the GTF file: /data/run/yoyopikachu/Sanju/ref_seq_annotation/Pvirgatum_516_v5.1.gene.gff3</span>
<span>Solution: check the formatting of the GTF file, it must contain some lines with exon in the 3rd column.</span>
<span>     Make sure the GTF file is unzipped.</span>
<span>     If exons are marked with a different word, use --sjdbGTFfeatureExon .</span></p><p><span>Nov 16 16:54:33 ...... FATAL ERROR, exiting</span>
<span>Mon Nov 16 16:54:33 EST 2020</span></p><ul><li><u><span>Solved</span></u><span>: used the new gff3 file &#39;Pvirgatum_516_v5.1.gene_exons.gff3&#39;</span></li></ul></blockquote><p>&nbsp;</p></li><li><p><span>Camelina had high mapped to multiple loci rate (&gt; 50%) in all camelina samples. Therefore, re-do alignment with all aligned read outputted and check what are the reads aligned to. </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Run another alignment with "--outSAMmapqUnique 0" to output all aligned reads (default:255 means only output unique reads)</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># This won't change anything except it outputs all aligned reads into .SAM file</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. STAR_map_camelina_multiple_check.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Sorting</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. sort_multi_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Count</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># '--nonunique all'; '-a 0'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae multi_mapped]<span class="cm-def">$ </span>. HTseq_count_multi.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 253px;"></div><div class="CodeMirror-gutters" style="display: none; height: 253px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Switchgrass had mapped to multiple loci rate around 25% in all samples. Also re-do alignment with all aligned read outputted and check what are the reads aligned to. </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Run another alignment with "--outSAMmapqUnique 0" to output all aligned reads (default:255 means only output unique reads)</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass]<span class="cm-def">$ </span>. STAR_map_multi_sg.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Sorting</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae multi_mapped]<span class="cm-def">$ </span>. sort_multi_sg.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Count</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># '--nonunique all'; '-a 0'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae multi_mapped]<span class="cm-def">$ </span>. HTseq_count_multi_sg.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 230px;"></div><div class="CodeMirror-gutters" style="display: none; height: 230px;"></div></div></div></pre></li></ol><p>&nbsp;</p><p><strong><span>December 03, 2020</span></strong></p><ol start='' ><li><p><span>Write script to extract primary transcript</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/extract_primary_transcripts_ID.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1.cds_primaryTranscriptOnly.fa <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1_primary_transcript_id.txt</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/extract_primary_transcripts_for_gff.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1.gene_exons.gff3 <span class="cm-attribute">-r</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1_primary_transcript_id.txt <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1_primary_transcript.gff3</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 391px;"></div><div class="CodeMirror-gutters" style="display: none; height: 391px;"></div></div></div></pre><blockquote><p><span>Number of gene in new gff3 matchs number of primary transcript</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu$ grep -c 'mRNA' /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1_primary_transcript.gff3</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">80278</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu$ grep -c '&gt;' /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1.cds_primaryTranscriptOnly.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">80278</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 184px;"></div><div class="CodeMirror-gutters" style="display: none; height: 184px;"></div></div></div></pre></blockquote><p>&nbsp;</p></li><li><p><span>Camelina sativa genome is reported as hexaploid </span></p><p><a href='https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7144077/pdf/1297.pdf' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7144077/pdf/1297.pdf</a></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># --outMultimapperOrder Random option outputs multiple alignments for each read in random order</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># re-map using '--outMultimapperOrder Random'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ cp</span> STAR_map_camelina.run STAR_map_camelina_random_assign.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. STAR_map_camelina_random_assign.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 92px;"></div><div class="CodeMirror-gutters" style="display: none; height: 92px;"></div></div></div></pre><blockquote><p><span>	</span><span>For multi-mappers, all alignments except one are marked with 0x100 (secondary alignment) in the FLAG (column 2 of the SAM). The unmarked alignment is selected from the best ones (i.e. highest scoring). This default behavior can be changed with </span><code>--outSAMprimaryFlag AllBestScore</code><span> option, that will output all alignments with the best score as primary alignments (i.e. 0x100 bit in the FLAG unset). </span></p><p><span>	</span><span>By default, the order of the multi-mapping alignments for each read is not truly random. The </span><code>--outMultimapperOrder Random</code><span> option outputs multiple alignments for each read in random order,  and also also randomizes the choice of the primary alignment from the highest scoring alignments. Parameter </span><code>--runRNGseed</code><span> can be used to set the random generator seed. With this option, the ordering of multi-mapping alignments of each read, and the choice of the primary alignment will vary from run to run, unless only one thread is used and the seed is kept constant. </span></p><p><span>	</span><span>The </span><code>--outSAMmultNmax parameter</code><span> limits the number of output alignments (SAM lines) for multimappers. For instance, </span><code>--outSAMmultNmax 1</code><span> will output exactly one SAM line for each mapped read. Note that NH:i: tag in STAR will still report the actual number of loci that the reads map to, while the the number of reported alignments for a read in the SAM file is min (NH,</span><code>--outSAMmultNMax</code><span>). </span></p></blockquote><p>&nbsp;</p></li><li><p><span>Re-run alignment- Switchgrass</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># using 'Pvirgatum_516_v5.1_primary_transcript.gff3'</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass_primary_transcript]<span class="cm-def">$ </span>. Genome_index.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass_primary_transcript]<span class="cm-def">$ </span>. STAR_map.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 69px;"></div><div class="CodeMirror-gutters" style="display: none; height: 69px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Re-run alignment for both Camelina and Switchgrass (Dec. 09, 2020)</span></p><p><span>Both Camelina and Switchgrass are polyploid, therfore, we decided to use </span><code>--outMultimapperOrder Random</code><span> option and only output one read by setting </span><code>--outSAMmultNmax 1</code><span> option. </span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## '--outMultimapperOrder Random' </span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## '--outSAMmultNmax 1'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## These setting don't affect alignment results in '*_Log.final.out'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Camelina</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. Genome_index_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. STAR_map_camelina_09Dec20.run <span class="cm-comment"># only Agri </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ cp</span> STAR_map_camelina_09Dec20.run STAR_map_camelina_09Dec20_coal_gyp.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. STAR_map_camelina_09Dec20_coal_gyp.run <span class="cm-comment"># coalmine and gypsum</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass_primary_transcript]<span class="cm-def">$ </span>. Genome_index.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae switchgrass_primary_transcript]<span class="cm-def">$ </span>. STAR_map_09Dec20.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 322px;"></div><div class="CodeMirror-gutters" style="display: none; height: 322px;"></div></div></div></pre><blockquote><p><span>Somehow there were errors popped up for both camelina and switchgrass. Therefore, I ran &#39;Genome_index.run&#39; again for both species. After re-ran, there was no error.</span></p></blockquote><p>&nbsp;</p></li><li><p><span>Sorting and counting aligned reads</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## copy scripts from original camelina/switchgrass directory</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## '--nonunique all' and '-a 0' still been used in HTseq-count since the flag in .SAM file should still be "multiple alignment". But there should be only one output alignments (SAM lines) for multimappers. </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># camelina</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ cp</span> sort_multi_camelina.run final_alignment/sort_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae multi_mapped]<span class="cm-def">$ cp</span> HTseq_count_multi.run ../final_alignment/HTseq_count.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ </span>. sort_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ </span>. HTseq_count.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae multi_mapped]<span class="cm-def">$ cp</span> sort_multi_sg.run ../../switchgrass_primary_transcript/final_alignment/sort_by_pos_sg.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae multi_mapped]<span class="cm-def">$ cp</span> HTseq_count_multi_sg.run ../../switchgrass_primary_transcript/final_alignment/HTseq_count_sg.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ </span>. sort_by_pos_sg.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ </span>. HTseq_count_sg.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Using '--nonunique none' and '-a 10' in one camelina and one switchgrass for testing.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># The results suggested that '--nonunique all' and '-a 0' are necessary.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># camelina</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ htseq</span><span class="cm-attribute">-count</span> <span class="cm-attribute">-f</span> bam <span class="cm-attribute">-r</span> pos <span class="cm-attribute">-i</span> Parent <span class="cm-attribute">-s</span> reverse <span class="cm-attribute">--nonunique</span> none <span class="cm-attribute">-m</span> union <span class="cm-attribute">-t</span> exon <span class="cm-attribute">-a</span> <span class="cm-number">10</span> camelina_Agri_1_sorted.bam /data/run/yoyopikachu/Sanju/ref_seq_annotation/Camelina_sativa.Cs.48.chr.gff3 &gt; count_camelina_Agri1_test_reverse.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ htseq</span><span class="cm-attribute">-count</span> <span class="cm-attribute">-f</span> bam <span class="cm-attribute">-r</span> pos <span class="cm-attribute">-i</span> Parent <span class="cm-attribute">-s</span> reverse <span class="cm-attribute">--nonunique</span> none <span class="cm-attribute">-m</span> union <span class="cm-attribute">-t</span> exon <span class="cm-attribute">-a</span> <span class="cm-number">10</span> SG-Ag-Up-40-1_sorted.bam /data/run/yoyopikachu/Sanju/ref_seq_annotation/Pvirgatum_516_v5.1_primary_transcript.gff3 &gt; count_switchgrass_SG-Ag-Up-40-1_test_reverse.txt</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 736px;"></div><div class="CodeMirror-gutters" style="display: none; height: 736px;"></div></div></div></pre><blockquote><ul><li><span>Camelina</span></li></ul><p><span>2635368 Dec 10 00:58 count_camelina_Agri_1_multi_RandomOutputOne.txt</span></p><p><span>2601227 Dec 10 00:55 count_camelina_Agri1_test_reverse.txt</span></p><p><img src="/Users/yoyopikachu/Documents/Note/Analysis_note/graph_for_note/Screen Shot 2020-12-10 at 5.16.26 PM.png" referrerpolicy="no-referrer" alt="Screen Shot 2020-12-10 at 5.16.26 PM"></p><ul><li><span>Switchgrass</span></li></ul><p><span>2068814 Dec 10 00:55 count_switchgrass_SG-Ag-Up-40-1_test_reverse.txt</span></p><p><span>2077519 Dec 10 00:52 count_SG-Ag-Up-40-1_multi_RandomOutputOne.txt</span></p><p>&nbsp;</p></blockquote><p>&nbsp;</p></li><li><p><span>Found there&#39;re non-primary transcripts in Camelina annotation file, therefore, downloaded all cds sequences from EnsemblPlant and, parsed and save only primary transcript.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Obtain primary transcript from cds.all.fa using python script</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of cds input:94352</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of primary cds:89275</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:Camelina_switchgrass yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/get_primary_from_cds.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs.cds.all.fa <span class="cm-attribute">-o1</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs.cds_primaryTranscriptOnly.fa <span class="cm-attribute">-o2</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_primany_transcript_id.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Comparing the output with output from other script (Camelina_sativa.Cs.pep.primary.fa)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Found 94352 accessions, 89275 genes, 0 unidentified transcripts</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Wrote 89275 genes</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:Camelina_switchgrass yoyopikachu<span class="cm-def">$ for</span> f <span class="cm-keyword">in</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs.pep.all.fa ; <span class="cm-keyword">do</span> python /Users/yoyopikachu/Documents/Lab/Comparative_genomics/OrthoFinder/tools/primary_transcript.py <span class="cm-def">$f</span> ; <span class="cm-keyword">done</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 483px;"></div><div class="CodeMirror-gutters" style="display: none; height: 483px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Extracted annotation for primary transcripts from the original annotation file.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># No. of mRNA in the original file: 90,321</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># No. of mRNA in new file: 85,151</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:Camelina_switchgrass yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/extract_primary_transcripts_for_gff_camelina.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs.48.chr.gff3 <span class="cm-attribute">-r</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_primany_transcript_id.txt <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs_primary_transcript.gff3</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:Camelina_switchgrass yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/extract_primary_transcripts_for_gff_camelina.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Downloads/Camelina_sativa.Cs.49.gff3 <span class="cm-attribute">-r</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_primany_transcript_id.txt <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs_primary_transcript_nonChr.gff3</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 437px;"></div><div class="CodeMirror-gutters" style="display: none; height: 437px;"></div></div></div></pre><blockquote><p><span>There are two annotation files for camelina on EnsemblPlant Camelina_sativa.Cs.49.gff3 and Camelina_sativa.Cs.49.chr.gff3</span></p><figure><table><thead><tr><th>&nbsp;</th><th><span>Camelina_sativa.Cs.49.gff3</span></th><th><span>Camelina_sativa.Cs.49.chr.gff3</span></th></tr></thead><tbody><tr><td><span>size</span></td><td><span>210.6 MB</span></td><td><span>200.3 MB</span></td></tr><tr><td><span>no. of mRNA</span></td><td><span>94,519</span></td><td><span>90,321</span></td></tr><tr><td><span>no. of primary mRNA</span></td><td><span>89,275</span></td><td><span>85,151</span></td></tr></tbody></table></figure><ul><li><span>Camelina_sativa.Cs.49.gff3 is the one we should use since the number of primary cds matches the number of number of primary mRNA in GFF file.</span></li></ul></blockquote><p>&nbsp;</p></li><li><p><span>Re-run alignment using the updated GFF file </span><em><span>Camelina_sativa.Cs_primary_transcript_nonChr.gff3</span></em></p><ul><li><span>Only primary transcripts are included in the updated GFF file</span></li><li><span>The number of primary mRNA matches the number of primary transcript in cds file.</span></li></ul><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># alignment</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. Genome_index_camelina_update_gff.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae camelina]<span class="cm-def">$ </span>. STAR_map_camelina_update_gff_14Dec20.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># copy script from previous</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ cp</span> sort_camelina.run ../final_align_update_gff/sort_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_alignment]<span class="cm-def">$ cp</span> HTseq_count.run ../final_align_update_gff/HTseq_count.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># sort and get read count</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_align_update_gff]<span class="cm-def">$ </span>. sort_camelina.run</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae final_align_update_gff]<span class="cm-def">$ </span>. HTseq_count.run</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 230px;"></div><div class="CodeMirror-gutters" style="display: none; height: 230px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Read count files were downloaded and differentially expressed analysis was conducted in local terminal using R.</span></p><ul><li><span>&quot;/Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Camelina_DESeq2/Note_Sanju_camelina.R&quot;</span></li><li><span>&quot;/Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Switchgrass_DESeq2/Note_Sanju_switchgrass.R&quot;</span></li><li><span>Raw reads were normalized using using &quot;median of ratios method&quot; of normalization. (</span><a href='https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html' target='_blank' class='url'>https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html</a><span>)</span></li></ul></li></ol><p>&nbsp;</p><p><strong><span>BLAST peptide between camellia/switchgrass and arabidopsis</span></strong></p><p><span>The interested gene lists that Sanju provided are Arabidopsis gene. Therefore, BLAST peptide between species is required in order to find out what are corresponded genes in camelina and switchgrass.</span></p><ol start='' ><li><p><span>Sequences download</span></p><ul><li><span>Camelina peptide was downloaded from EnsenblPlants (</span><a href='http://plants.ensembl.org/Camelina_sativa/Info/Index' target='_blank' class='url'>http://plants.ensembl.org/Camelina_sativa/Info/Index</a><span>)</span></li><li><span>Switchgrass peptide was downloaded from JGI Phytozome 12 (</span><a href='https://phytozome.jgi.doe.gov/pz/portal.html#!bulk?org=Org_Pvirgatum' target='_blank' class='url'>https://phytozome.jgi.doe.gov/pz/portal.html#!bulk?org=Org_Pvirgatum</a><span>)</span></li><li><span>Arabidopsis peptide was downloaded from TAIR10 database (</span><a href='https://www.arabidopsis.org/download/index-auto.jsp?dir=%2Fdownload_files%2FSequences%2FTAIR10_blastsets' target='_blank' class='url'>https://www.arabidopsis.org/download/index-auto.jsp?dir=%2Fdownload_files%2FSequences%2FTAIR10_blastsets</a><span>)</span></li></ul><p>&nbsp;</p></li><li><p><span>Extract primary pep in camelina</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Obtain primary transcript from cds.all.fa using python script</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of pep input:94352</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of primary allele:89275</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of pep output:71117</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># The difference was due to peptide ID problem. Some of the peptide IDs include ".cds1" such as "Csa02g061700.1.cds1".</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/extract_primary_pep_by_given_name.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs.pep.all.fa <span class="cm-attribute">-r</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_primany_transcript_id_for_pep.txt <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs_primary_pep.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Gererate primary peptide using another script</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of peptide input:94352</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Number of primary peptide:89275</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">python3 /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/get_primary_from_pep.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Camelina_sativa.Cs.pep.all.fa <span class="cm-attribute">-o1</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/test_camelina_primary_pep.txt <span class="cm-attribute">-o2</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/test_camelina_primary_pep_id.txt</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 644px;"></div><div class="CodeMirror-gutters" style="display: none; height: 644px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>BLAST peptide</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Build database for blast</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae BLAST_pep]<span class="cm-def">$ module</span> load BLAST/2.2.26</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae BLAST_pep]<span class="cm-def">$ formatdb</span> <span class="cm-attribute">-i</span> /data/run/yoyopikachu/Sanju/ref_seq_annotation/TAIR10_pep_20110103_representative_gene_model_updated.txt <span class="cm-attribute">-n</span> arab_peptide <span class="cm-attribute">-p</span> T <span class="cm-attribute">-t</span> arab_peptide</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># BLAST switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -m 8 option</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae BLAST_pep]<span class="cm-def">$ blastall</span> <span class="cm-attribute">-i</span> /data/run/yoyopikachu/Sanju/ref_seq_annotation/Pvirgatum_516_v5.1.protein_primaryTranscriptOnly.fa <span class="cm-attribute">-d</span> arab_peptide <span class="cm-attribute">-p</span> blastp <span class="cm-attribute">-e</span> 1e-10 <span class="cm-attribute">-b</span> <span class="cm-number">5</span> <span class="cm-attribute">-v</span> <span class="cm-number">5</span> <span class="cm-attribute">-m</span> <span class="cm-number">8</span> <span class="cm-attribute">-o</span> switcgrass_v_arabidopsis_pep.blast</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># default option (0 = pairwise)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae BLAST_pep]<span class="cm-def">$ blastall</span> <span class="cm-attribute">-i</span> /data/run/yoyopikachu/Sanju/ref_seq_annotation/Pvirgatum_516_v5.1.protein_primaryTranscriptOnly.fa <span class="cm-attribute">-d</span> arab_peptide <span class="cm-attribute">-p</span> blastp <span class="cm-attribute">-e</span> 1e-10 <span class="cm-attribute">-b</span> <span class="cm-number">5</span> <span class="cm-attribute">-v</span> <span class="cm-number">5</span> <span class="cm-attribute">-o</span> switcgrass_v_arabidopsis_pep.blast.out</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># BLAST camelina</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae BLAST_pep]<span class="cm-def">$ blastall</span> <span class="cm-attribute">-i</span> /data/run/yoyopikachu/Sanju/ref_seq_annotation/Camelina_sativa.Cs.pep.primary.fa <span class="cm-attribute">-d</span> arab_peptide <span class="cm-attribute">-p</span> blastp <span class="cm-attribute">-e</span> 1e-10 <span class="cm-attribute">-b</span> <span class="cm-number">5</span> <span class="cm-attribute">-v</span> <span class="cm-number">5</span> <span class="cm-attribute">-m</span> <span class="cm-number">8</span> <span class="cm-attribute">-o</span> camelina_v_arabidopsis_pep.blast</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># default option (0 = pairwise)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[yoyopikachu@orchidaceae BLAST_pep]<span class="cm-def">$ blastall</span> <span class="cm-attribute">-i</span> /data/run/yoyopikachu/Sanju/ref_seq_annotation/Camelina_sativa.Cs.pep.primary.fa <span class="cm-attribute">-d</span> arab_peptide <span class="cm-attribute">-p</span> blastp <span class="cm-attribute">-e</span> 1e-10 <span class="cm-attribute">-b</span> <span class="cm-number">5</span> <span class="cm-attribute">-v</span> <span class="cm-number">5</span> <span class="cm-attribute">-o</span> camelina_v_arabidopsis_pep.blast.out</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 575px;"></div><div class="CodeMirror-gutters" style="display: none; height: 575px;"></div></div></div></pre><blockquote><p><a href='http://www.pangloss.com/wiki/Blast' target='_blank' class='url'>http://www.pangloss.com/wiki/Blast</a></p><p><span>-m 8 option for tabular output header:</span></p><figure><table><thead><tr><th><span>query</span></th><th><span>subject</span></th><th><span>%id</span></th><th><span>alignment length</span></th><th><span>mismatches</span></th><th><span>gap openings</span></th><th><span>query start</span></th><th><span>query end</span></th><th><span>subject start</span></th><th><span>subject end</span></th><th><span>E value</span></th><th><span>bit score</span></th></tr></thead><tbody><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table></figure></blockquote><p>&nbsp;</p></li><li><p><span>Extract BLAST results</span></p><p><span>The threshold for gene similiarity is set to 80% coverage and 80% identity. For each BLAST result, only keep one top hit.</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## caculate peptide length (for calculating BLAST coverage)</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Desktop/count_peptide_length.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/Pvirgatum_516_v5.1.protein_primaryTranscriptOnly.fa <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/sg_peptide_length.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># camelina</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Desktop/count_peptide_length_camelina.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/test_camelina_primary_pep.txt <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/camelina_peptide_length.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Filter blastp results by 80% identity and 80% coverage</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># camelina</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 371589 blast results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 61004 genes with coverage and identity greater than or equal to 80</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Desktop/blast_filter_coverage_identity_camelina.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/BLAST/camelina_v_arabidopsis_pep.blast <span class="cm-attribute">-len</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/camelina_peptide_length.fa <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/BLAST/camelina_v_arabidopsis_pep_80_id_cover_blastp.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># switchgrass</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 203607 blast results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 4901 genes with coverage and identity greater than or equal to 80</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">Rose-MacBookPro:~ yoyopikachu<span class="cm-def">$ python3</span> /Users/yoyopikachu/Desktop/blast_filter_coverage_identity_camelina.py <span class="cm-attribute">-i</span> /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/BLAST/switcgrass_v_arabidopsis_pep.blast <span class="cm-attribute">-len</span> /Users/yoyopikachu/Documents/Lab/Comparative_genomics/Ref_genomes/Camelina_switchgrass/sg_peptide_length.fa <span class="cm-attribute">-o</span> /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/BLAST/switchgrass_v_arabidopsis_pep_80_id_cover_blastp.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 920px;"></div><div class="CodeMirror-gutters" style="display: none; height: 920px;"></div></div></div></pre></li></ol><p>&nbsp;</p><p><strong><span>Compiling the results with Sanju&#39;s gene lists</span></strong></p><ol start='' ><li><p><span>Parse gene files that provided by Sanju</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: currentcolor;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: currentcolor;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Lipid-related gene: 883</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Mineral and metal transporter gene: 55</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Starch synthesis and degradation gene: 59</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Mainly compiled in Sublime Text generate </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">/Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/gene_list_ready/All_lipid_v2.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">/Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/gene_list_ready/all_mineral_metal_transporter_gene_v2</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">/Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/interested_genes/gene_list_ready/all_starch_synthesis_degrade</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 253px;"></div><div class="CodeMirror-gutters" style="display: none; height: 253px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Combine BLAST results, gene groups, DEG analysis and normalized read count</span></p><p><span>Camelina:</span></p><ul><li><span>Control vs coalmine: /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Camelina_DESeq2/Note_Sanju_functional_DEG_camelina_control_v_coal.R</span></li><li><span>Control vs gypsum: /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Camelina_DESeq2/Note_Sanju_functional_DEG_camelina_control_v_gypsum.R</span></li><li><span>Coalmine vs gypsum: /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Camelina_DESeq2/Note_Sanju_functional_DEG_camelina_coalmine_v_gypsum.R </span></li></ul><p><span>Switchgrass:</span></p><ul><li><span>Control vs coalmine (Trt): /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Switchgrass_DESeq2/Note_Sanju_functional_DEG_switchgrass_control_v_coal.R</span></li><li><span>Lowland vs upland (Geno): /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Switchgrass_DESeq2/Note_Sanju_functional_DEG_switchgrass_lowland_vs_upland.R</span></li><li><span>Interaction (Trt x Geno): /Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/count_random_output_one/Switchgrass_DESeq2/Note_Sanju_functional_DEG_switchgrass_TrtxGeno.R</span></li></ul><blockquote><p><span>For double-checking the final metadata, padj and log2foldchange of three genes, in each output files, have been randomly selected and compred with the values in original DESeq2 output. </span></p></blockquote><p>&nbsp;</p></li><li><p><span>Heatmap</span></p><ul><li><p><span>Heatmaps were made in R (/Users/yoyopikachu/Documents/Lab/Codon_usage/mRNAseq/Sanju/Note_Sanju_heatmap_all.R).</span></p><p>&nbsp;</p></li></ul></li><li><p><span>Prepare metadata files for Sanju</span></p><ul><li><span>Three duplicated columns (*_ID_check) were removed in all 18 metadata files.</span></li><li><span>Query and Hit were renamed as &quot;SG/CAM_gene_ID&quot; and &quot;AT_gene_ID&quot;.</span></li></ul></li><li><p><span>d</span></p></li></ol><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p></div>
</body>
</html>